SparkConnector/sparkconnector/configuration.py

```diff
@@ -132,11 +132,27 @@ def configure(self, opts, ports):
 
         # Extend conf adding logging of log4j to java options
         base_extra_java_options = "-Dlog4j.configuration=file:%s" % self.connector.log4j_file
-        extra_java_options = conf.get("spark.driver.extraJavaOptions")
-        if extra_java_options:
-            extra_java_options = base_extra_java_options + " " + extra_java_options
+        extra_java_options = None
+
+        # For customenv sessions, do not allow extraJavaOptions to be overwritten by the extra_java_options, 
+        # since the NXCals instances, coming as a default value by the SparkContext when extraJavaOptions is
+        # not specified, need to be preserved in the driver configuration.
+        if os.environ.get("SOFTWARE_SOURCE") == "customenv":
+            SPARK_DEFAULTS = os.environ.get(
```

Comment:
Here, first line inside the if, I would initialize `extra_java_options = ''`. See why below.